app/src/main/java/com/amaze/filemanager/ui/activities/superclasses/ThemedActivity.java

```diff
@@ -85,27 +82,6 @@ public void onCreate(Bundle savedInstanceState) {
     setTheme();
   }
 
-  @Override
```

Comment:
This was removed because it was causing state to be lost when orientation changed, or activity recreated for any other reason.

Seems like nothing else changed with this change, app runs fine if the theme is not set or set to default.

Reply:
What? That annotation doesn't really do anything AFAIK, the override is created nontheless.

Reply:
I meant the methods below the annotation, onConfigurationChanged and restartPC. This is just the first line

Reply:
> Seems like nothing else changed with this change, app runs fine if the theme is not set or set to default.

Does it still correctly set the theme immediately (in the settings screen) on theme change?

Reply:
> Seems like nothing else changed with this change, app runs fine if the theme is not set or set to default.

Changing the system theme whilst using the app (with theme set to follow system) doesn't work as expected. The app theme stays intact until the app is restated.

> This was removed because it was causing state to be lost when orientation changed, or activity recreated for any other reason.

Can you revert this change and handle the above condition in some other way?

Reply:
In this case, I will try to handle it more gracefully than before. The way it was if the theme was set to follow the system, every configuration change (such as orientation) would cause the activity to be restarted (not recreated, restarted as in a new activity is created from the ground up), if I understand correctly.

This is bad because it messes up with saved bundle badly, which can cause lots of frustration when you are trying to make fragments work and they just reset whenever orientation changes. There are some hacky solutions which were being used before but it is not elegant at all.

I believe the best alternative is instead to call `Activity` `recreate` whenever the system theme changes and we should use the same theme as the system. This causes a screen flicker, but still IMO is a better solution.

Reply:
Great! I should've done that in the first place whilst implementing too 🙈